pkg/controllers/hub/trafficmanagerprofile/controller.go

```diff
@@ -95,35 +112,59 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		return r.handleDelete(ctx, profile)
 	}
 
+	// register metrics finalizer
+	if !controllerutil.ContainsFinalizer(profile, objectmeta.MetricsFinalizer) {
```

Comment:
Can we just reuse the existing TrafficManagerProfileFinalizer?

Reply:
nope, we cannot, it's explained in line 137